### PR TITLE
Add a check for multiplex op

### PR DESCRIPTION
**Pull request by aoyulong**
### PR types
Others

### PR changes
Ops

### Describe
The paddle.multiplex will break down when the size of candidate tensors is zero and doesn't give useful an error message. This pr will check the situation and gives user more information.

### CODE REVIEW

File: paddle/fluid/operators/multiplex_op.cu [hunk 1]
```diff
@@ -29,6 +29,10 @@ class MultiplexGPUKernel : public framework::OpKernel<T> {
     auto* out = ctx.Output<Tensor>("Out");
     out->mutable_data<T>(ctx.GetPlace());
 
+    PADDLE_ENFORCE_GT(ins[0]->numel(), 0,
```
Comment (ForFishes) [inline paddle/fluid/operators/multiplex_op.cu]: Why only check the first tensor of the input? Shouldn't it be checked more than one becasue of MultiInput type?

File: paddle/fluid/operators/multiplex_op.h [hunk 2]
```diff
@@ -31,6 +31,10 @@ class MultiplexCPUKernel : public framework::OpKernel<T> {
 
     out->mutable_data<T>(ctx.GetPlace());
 
+    PADDLE_ENFORCE_GT(ins[0]->numel(), 0,
```
Comment (ForFishes) [inline paddle/fluid/operators/multiplex_op.h]: Same as above.

File: paddle/fluid/operators/multiplex_op.h [hunk 3]
```diff
@@ -31,6 +31,10 @@ class MultiplexCPUKernel : public framework::OpKernel<T> {
...
+                      platform::errors::PreconditionNotMet(
+                          "indexing will be out of bounds with size 0."));
```
Comment (ForFishes) [inline paddle/fluid/operators/multiplex_op.h]: indexing->index ? PreconditionNotMet-> OutOfRange?

Comment (aoyulong) [reply paddle/fluid/operators/multiplex_op.h]: Done.

File: paddle/fluid/operators/multiplex_op.h (hunk 2 above)
Comment (aoyulong) [reply paddle/fluid/operators/multiplex_op.h]: Done.

File: paddle/fluid/operators/multiplex_op.cu (hunk 1 above)
Comment (aoyulong) [reply paddle/fluid/operators/multiplex_op.cu]: Done.

**Review by ForFishes — APPROVED** [review]
LGTM

**Pull request merged**